# Genomic analysis of Stalk-eyed flies 

[Stalkie_GoogleDoc](https://docs.google.com/document/d/1TT5HuJuPho5OCNKg7C28hcHOtwxHuN2BFG5LlgKJFnE/edit)

Teleopsis dalmanni

Karyotype (like Drosophila): 3 chromosomes

~620Mb genome size

SR driving X-chromosomes found in 20-30% of wild indivs ([Reinhardt et al. 2014](https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004362))

Sympatric cryptic species in the same area. 


Meiotic drive is ancestral but lost in the sister species

![alt_txt][Fig1]

[Fig1]:https://user-images.githubusercontent.com/12142475/136238893-1272665f-85ba-4eab-82ef-0924ef1b529b.png


## People 

[POM](https://www.ucl.ac.uk/biosciences/people/andrew-pomiankowski) - PI

[Sasha Bradshaw](https://london-nerc-dtp.org/profile/bradshaws/) - PhD student with POM

[Flo Camus](http://www.homepages.ucl.ac.uk/~ucbtmre/Labsite/Flo.html) - Drosophila biologist, mtDNA, UCL

[Helen Hipperson](https://www.sheffield.ac.uk/biosciences/people/academic-staff/helen-hipperson) - Sheffield. Leading analyses of genomic data

[Christophe Dessimoz](https://lab.dessimoz.org/people/christophe-dessimoz) - Uni Lausanne & SIB. [OMA database of orthologs](https://omabrowser.org/oma/about/) - potentially for annotation. 


## Available data

### 2 genomes available: 

1) from inbred males with SR genotype (based on genetic markers - not msats). 

PacBio sequences + HiC. Assembled by Phase Genomics

2) Wild type (ST) genome being generated in Liverpool (Terry Burke and Helen Hipperson). 

Raw data available (PacBio HiFi reads x 2 cells). Needs assembly

```
zcat m64157e_211024_013127.hifi_reads.fasta.gz |grep -v ">" |awk '{x+=length($0)}END{print x}'

7893963766

zcat m64157e_210730_141553.hifi_reads.fasta.gz |grep -v ">" |awk '{x+=length($0)}END{print x}'
12769157425

Total = ~20Billion bases. Expected genome size = 438Mb

Expected haploid coverage = 20Bil/438Mb = 45x
```


A genome has been assembled by Jerry Wilkinson in 2020: [BioRxiv paper](https://www.biorxiv.org/content/10.1101/2020.09.23.310227v1.full)


### 100 WGS wild flies

50:50 SR:wild (based on genetic markers, but genetic markers were developed in inbred lab strains, so we're uncertain if all 50 are def SR). 

sampled over ~10 years within ~10km from each other. 


## Theories and questions: 

#### 1. Series of overlapping inversions?

US group working on Stalk-eyed flies: Gerald Wilkinson (Uni Maryland) 

-> Found a series of inversions (4-5; overlapping) on the X-chromosome that they think are linked to SR meiotic drive. 

1) Can we recover the same inversions? Do the inversions overlap?  (We'll be using a newly assembled genome 1 above, which is based on a more homozygous sample than the reference genome generated by UniMaryland)


#### 2. Are inversions in coding regions?

Inversions are likely large based on microsat data. 

2) Can we find the break points of the inversions? Are they in coding regions? 

(Need an annotated genome)

InvBFM


#### 3. Are inversions adaptive? 

There is an ~20% inversion rate in the wild. Recombination in these individuals will be reduced compared to the rest of the pop, so we expect an accumulation of mutations. 

Thus we expect these males to have a high mutational load and be poorer quality. BUT they have the same fecundity as normal flies. 

They have a smaller gap between the eyes, which is less attractive to females. Thus they have fewer mating opportunites. They don't deliver more sperm per insemination. But they have much bigger testes. 


3) Do we find an accumulation of mutations in the SR males? 

Estimate mutational load in each dataset. (for different polymorphism classes- SNPs, indels, inversions?)

Fst outlier analysis to find SNP differences between datasets. Is there more diversity within the SR males than between groups? 



#### 4. mtDNA?

Can we extract the mtDNA from the WGS dataset? And from the reference genome? 

Greg Hurst and Stefanos: mitogenome



## Possible tools

[InvBFM](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-6585-1): Machine learning approach using features collected from several SV analyses to predict locations of SV. 


## Genome assembly and annotation

[Howe et al. (2021) Significantly improving the quality of genome assemblies through curation. Gigascience](https://academic.oup.com/gigascience/article/10/1/giaa153/6072294?login=true)

Paper outlining the recommended genome assembly, checking, and polishing pipeline as used by DToL and others. 


Check genome quality: 

1) Proportion of BUSCO genes represented

2) Proportion BUSCO genes in unscaffolded contigs

3) Map genome to a) sister species b) draft of the same species

4) What stats? 



## Pipeline

Pipeline used for assembling the genome and checking the quality based on the Howe et al. 2021 recommendations. 

See the DToL [Pararge aegeria genome note](https://d212y8ha88k086.cloudfront.net/manuscripts/19102/f440c634-7f02-4e43-a5aa-febe4b9336f7_17278_-_jonathan_threlfall.pdf?doi=10.12688/wellcomeopenres.17278.1&numberOfBrowsableCollections=9&numberOfBrowsableInstitutionalCollections=0&numberOfBrowsableGateways=14) for an example of their pipeline. 

### 1. Raw Data 

These checks can be done at the start with the raw data or after assembly to remove contaminants.

1. Check for contaminants (vectors, adapters, organelles)

2. Assemble mtDNA genome using Organelle-PBA specifically designed for PacBio data or [MitoHiFi](https://github.com/marcelauliano/MitoHiFi) designed for HiFi reads. 

Organelle_PBA paper [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5219736/); and software [here](https://github.com/aubombarely/Organelle_PBA) 

Or map raw reads to an available mitogenome (sister species or same species, or combine all available references in a single reference fasta file). Use [minimap2](https://github.com/lh3/minimap2) to map and extract the scaffolds that match the reference mitogenomes. 

### 2. Automated assembly


1. Assemble HiFi reads using [HiFiasm](https://github.com/chhylp123/hifiasm#limit)

This is the recommended software given HiFi reads. See preprint [here](https://www.biorxiv.org/content/10.1101/2022.02.15.480579v2) for a comparison of assembler performance. 

*"We found that centromeres and rDNA clusters are responsible for 71% of contig breaks in the CLR scaffolds, while relatively short stretches of GA/TC repeats are at the core of >85% of the unfilled gaps in our best HiFi assemblies."*

2. Haplotypic duplications identified and removed using [purge_dups](https://github.com/dfguan/purge_dups). 

3. Contaminants removed ([Blobtoolkit](https://github.com/blobtoolkit/blobtoolkit))

4. Trailing Ns removed 

5. Kmer spectrum fits expectations 

6. Haplotypes correctly assorted (if fully haplotype resolved) 

7. Polishing reliable (NUMTS/NUPTS preserved and frameshifts corrected). 

    - One round of polishing where the genome is mapped against 1) a draft of the same species, 2) sister species, 3) SNPs called. 

8. Organelle genomes as unintended byproduct of assembly identified and removed (see MitoHiFi above)

## 3. Cleaned assembly

1. All available raw data aligned, discrepancies marked up

2. All available data from other sources aligned. Discrepancies marked up

3. Discrepancies assessed and corrected where necessary

    a) misjoins broken
    
    b) missing joins made
    
    c) inversions reoriented
    
    d) false duplications removed
    
    
## 4. Structural corrections

1. Crhomosome level units identified and named

2. Sex chromosomes identified and named

3. unlocalised units identified and named

4. organelle genomes named. 

## 5. Submission to archive
